cmd/erigon-el/backend/backend.go

```diff
@@ -750,19 +750,35 @@ func (s *Ethereum) StartMining(ctx context.Context, db kv.RwDB, mining *stagedsy
 		newHeadCh, closeNewHeadCh := s.notifications.Events.AddHeaderSubscription()
 		defer closeNewHeadCh()
 
+		log.Info("Starting to mine", "etherbase", eb)
+
 		var works bool
 		var hasWork bool
 		errc := make(chan error, 1)
 
+		// create an empty channel to immediately start mining
+		first := make(chan struct{}, 1)
+		first <- struct{}{}
+		defer close(first)
+
 		for {
-			mineEvery.Reset(cfg.Recommit)
+			// Only reset if some work was done previously
+			if hasWork {
+				mineEvery.Reset(cfg.Recommit)
+			}
 			select {
```

Comment:
Couldn't you put the select in an if statement of `if !hasWork { select ... }`  and then start up with `hasWork = true`? so you only go in the select if you need to wait for work otherwise you can skip it ( which will only happen at the beginning )